PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -2283,26 +2708,242 @@ Bool_t MaxNumberOfEvents()
+                          continue;
+                        }
+                        Double_t dPhi11 = nl_a.ftaNestedLoops[0]->GetAt(i11);
+                        Double_t dW11 = nl_a.ftaNestedLoops[1]->GetAt(i11);
+                        if (11 == order) {
+                          value = TMath::Cos(harmonics->GetAt(0) * dPhi1 + harmonics->GetAt(1) * dPhi2 + harmonics->GetAt(2) * dPhi3 + harmonics->GetAt(3) * dPhi4 + harmonics->GetAt(4) * dPhi5 + harmonics->GetAt(5) * dPhi6 + harmonics->GetAt(6) * dPhi7 + harmonics->GetAt(7) * dPhi8 + harmonics->GetAt(8) * dPhi9 + harmonics->GetAt(9) * dPhi10 + harmonics->GetAt(10) * dPhi11);
+                          weight = dW1 * dW2 * dW3 * dW4 * dW5 * dW6 * dW7 * dW8 * dW9 * dW10 * dW11;
+                          profile->Fill(0.5, value, weight);
+                          continue;
+                        }
+                        for (int i12 = 0; i12 < nParticles; i12++) {
+                          if (i12 == i1 || i12 == i2 || i12 == i3 || i12 == i4 || i12 == i5 || i12 == i6 || i12 == i7 || i12 == i8 || i12 == i9 || i12 == i10 || i12 == i11) {
+                            continue;
+                          }
+                          Double_t dPhi12 = nl_a.ftaNestedLoops[0]->GetAt(i12);
+                          Double_t dW12 = nl_a.ftaNestedLoops[1]->GetAt(i12);
+                          if (12 == order) {
+                            value = TMath::Cos(harmonics->GetAt(0) * dPhi1 + harmonics->GetAt(1) * dPhi2 + harmonics->GetAt(2) * dPhi3 + harmonics->GetAt(3) * dPhi4 + harmonics->GetAt(4) * dPhi5 + harmonics->GetAt(5) * dPhi6 + harmonics->GetAt(6) * dPhi7 + harmonics->GetAt(7) * dPhi8 + harmonics->GetAt(8) * dPhi9 + harmonics->GetAt(9) * dPhi10 + harmonics->GetAt(10) * dPhi11 + harmonics->GetAt(11) * dPhi12);
+                            weight = dW1 * dW2 * dW3 * dW4 * dW5 * dW6 * dW7 * dW8 * dW9 * dW10 * dW11 * dW12;
+                            profile->Fill(0.5, value, weight);
+                            continue;
+                          }
+
+                          // ... it's easy to continue the above pattern here
+
+                        } // for(int i12=0; i12<nParticles; i12++)
+                      }   // for(int i11=0; i11<nParticles; i11++)
+                    }     // for(int i10=0; i10<nParticles; i10++)
+                  }       // for(int i9=0; i9<nParticles; i9++)
+                }         // for(int i8=0; i8<nParticles; i8++)
+              }           // for(int i7=0; i7<nParticles; i7++)
+            }             // for(int i6=0; i6<nParticles; i6++)
+          }               // for(int i5=0; i5<nParticles; i5++)
+        }                 // for(int i4=0; i4<nParticles; i4++)
+      }                   // for(int i3=0; i3<nParticles; i3++)
+    }                     // for(int i2=0; i2<nParticles; i2++)
+  }                       // for(int i1=0; i1<nParticles; i1++)
```

Comment:
You are aware of the CPU cost of these nested loops, right?

Reply:
Yes. They are used to cross-check for small multiplicity (M <= 15) independent analysis formulas in terms of Q-vectors.